### PR TITLE
Implements offline player location API

### PR DESCRIPTION
**Pull request by Faithcaio**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/1787) | **SpongeCommon**

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java [hunk 1]
```diff
@@ -24,10 +24,12 @@
  */
 package org.spongepowered.common.mixin.core.entity.player;
 
+import static ca.weblite.objc.RuntimeUtils.loaded;
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java]: what no

**Review by kashike — DISMISSED** [review]

File: src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java [hunk 2]
```diff
@@ -1252,4 +1256,22 @@ public void remove() {
...
+    public void setLocation(Vector3d position, UUID world) {
+        Optional<WorldProperties> prop = Sponge.getServer().getWorldProperties(world);
+        if (!prop.isPresent()) {
+            throw new IllegalArgumentException("Invalid World: No world found for UUID");
+        }
+        Optional<World> loaded = Sponge.getServer().loadWorld(prop.get());
+        if (!loaded.isPresent()) {
+            throw new IllegalArgumentException("Invalid World: Could not load world for UUID");
+        }
+        this.setLocation(new Location<>(loaded.get(), position));
```
Comment (liach) [inline src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java]: Call `loaded.orElseThrow(() -> new IllegalArgumentException("Invalid World: Could not load world for UUID"))` instead

File: src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java [hunk 3]
```diff
@@ -1252,4 +1256,22 @@ public void remove() {
...
+    @Override
+    public Optional<UUID> getWorldUniqueId() {
+        return Optional.of(this.getWorld().getUniqueId());
+    }
+
+    @Override
+    public void setLocation(Vector3d position, UUID world) {
+        Optional<WorldProperties> prop = Sponge.getServer().getWorldProperties(world);
+        if (!prop.isPresent()) {
+            throw new IllegalArgumentException("Invalid World: No world found for UUID");
```
Comment (liach) [inline src/main/java/org/spongepowered/common/mixin/core/entity/player/MixinEntityPlayerMP.java]: Use `WorldProperties prop = Sponge.getServer().getWorldProperties(world).orElseThrow(() -> new IllegalArgumentException("Invalid World: No world found for UUID"));` instead

File: src/main/java/org/spongepowered/common/data/util/NbtDataUtil.java [hunk 4]
```diff
@@ -416,4 +418,16 @@ public static NBTTagList newDoubleNBTList(double... numbers) {
         return nbttaglist;
     }
 
+    public static NBTTagList newFloatNBTList(float... numbers)
+    {
```
Comment (liach) [inline src/main/java/org/spongepowered/common/data/util/NbtDataUtil.java]: faith you need to run formatter after copy-paste

**Pull request merged**